### PR TITLE
Compile teensy

### PR DESCRIPTION
**Pull request by blhough**
The project now compiles on Teensy! 3.2 and 3.6 (Tested on a 3.2)

The Arduino IDE has a "feature" where it automatically tries to compile every .cpp and .h file in the same directory as the .ino file so I moved all the files that need to compile on the teensy into their own subdirectory  named "Micromouse".

I removed Main.cpp becuase Micromouse.ino now takes it place.

std I/O does not work on Teensy so all `#include <iostream>`, `<fstream>`, etc have been disabled when compiling on Teensy. Use `Serial.printf()` for cross platform printing.

SConstruct still needs updating.

### CODE REVIEW

**Comment by blhough** [comment]
@ids1024 I had to reorganize things to get it to work with the Arduino IDE. Can you update SConstruct?

File: Simulation/simulate.cpp [hunk 1]
```diff
@@ -1,10 +1,10 @@
 #include <SFML/Graphics.hpp>
 
-#include "Coordinate.h"
-#include "Direction.h"
-#include "Maze.h"
-#include "Node.h"
-#include "Path.h"
+#include "..\Micromouse\Coordinate.h"
```
Comment (ids1024) [inline Simulation/simulate.cpp]: On Linux, these need to be forward slashes (`/`), or it doesn't compile. Do those work on Windows?

If not, I don't know what the best way to achieve compatibility is.

File: Micromouse/Maze.cpp [hunk 2]
```diff
@@ -256,12 +261,12 @@ void Maze::findPath(CellCoordinate start, CellCoordinate end,
         }
     }
 
-    throw runtime_error("No path found");
```
Comment (ids1024) [inline Micromouse/Maze.cpp]: I think we want some error handling here (perhaps only in the simulator)

File: Micromouse/Maze.cpp [hunk 3]
```diff
@@ -256,12 +261,12 @@ void Maze::findPath(CellCoordinate start, CellCoordinate end,
...
+    //    throw runtime_error("No path found");
 }
 
 Node &Maze::getNode(NodeCoordinate pos) {
-    if (!withinBounds(pos))
```
Comment (ids1024) [inline Micromouse/Maze.cpp]: Again, it should print some error, at least in the simulator

**Review by ids1024 — CHANGES_REQUESTED** [review]

**Comment by ids1024** [comment]
Ok, I've pushed the `SConstruct`, and the needed fixes for it to build with scons (including the slash direction).

File: Simulation/simulate.cpp (hunk 1 above)
Comment (blhough) [reply Simulation/simulate.cpp]: Yes forward slashes work. The auto generated ones are back slashes though so im going to forget a lot.

**Pull request merged**